compiler/rustc_infer/src/infer/combine.rs

```diff
@@ -470,25 +470,17 @@ impl<'infcx, 'tcx> CombineFields<'infcx, 'tcx> {
             RelationDir::SupertypeOf => ty::Contravariant,
         };
 
-        trace!(?ambient_variance);
-
-        let for_universe = self.infcx.probe_ty_var(for_vid).unwrap_err();
-        let for_vid_sub_root = self.infcx.inner.borrow_mut().type_variables().sub_root_var(for_vid);
-
-        trace!(?for_universe);
-        trace!(?self.trace);
-
-        Generalizer {
-            infcx: self.infcx,
-            delegate: self,
+        generalize(
```

Comment:
nit: prefer `generalize::generalize` here instead of importing the method or importing `Generalizer` and changing it to `Generalizer::generalize`. makes it clear that `generalize` is from a different file